### ISSUE TITLE
Docker-in-Docker: Docker Compose failing to install on ubuntu:jammy

### ISSUE THREAD

**Issue by jacobwoffenden** [devcontainers/features#619]
Seeing

```
29.32 (*) Installing compose-switch...
30.02 compose_switch_version=1.0.5
30.41 curl: (60) SSL: no alternative certificate subject name matches target host name 'objects.githubusercontent.com'
30.41 More details here: https://curl.se/docs/sslcerts.html
[2023-07-19T16:38:37.334Z] 
30.41 
30.41 curl failed to verify the legitimacy of the server and therefore could not
30.41 establish a secure connection to it. To learn more about this situation and
30.41 how to fix it, please visit the web page mentioned above.
30.41 ERROR: Feature "Docker (Docker-in-Docker)" (ghcr.io/devcontainers/features/docker-in-docker) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/docker-in-docker for help troubleshooting this error.
```

when using 

```json
"ghcr.io/devcontainers/features/docker-in-docker:2": {}
```

The only way I can get `docker-in-docker` to install is when specifying `"dockerDashComposeVersion": "none"`

Maybe related to

> Important
> 
> From July 2023 Compose V1 stopped receiving updates. It’s also no longer available in new releases of Docker Desktop.
> 
> Compose V2 is included with all currently supported versions of Docker Desktop. For more information, see [Migrate to Compose V2](https://docs.docker.com/compose/migrate).
> 
> Docker’s documentation refers to and describes Compose V2 functionality.

source: https://docs.docker.com/compose/install/standalone/

**Comment by samruddhikhandale** [comment]
> Important
> 
> From July 2023 Compose V1 stopped receiving updates. It’s also no longer available in new releases of Docker Desktop.
> 
> Compose V2 is included with all currently supported versions of Docker Desktop. For more information, see [Migrate to Compose V2](https://docs.docker.com/compose/migrate).
> 
> Docker’s documentation refers to and describes Compose V2 functionality.

> source: 
https://docs.docker.com/compose/install/standalone/


I agree, and I think we should switch the default for `dockerDashComposeVersion` Feature option to `v2`, thanks for pointing that out.

**Comment by jacobwoffenden** [comment]
I'm just preparing a PR, but was caught out by https://bugs.launchpad.net/ubuntu/+source/curl/+bug/2028170

**Comment by jacobwoffenden** [comment]
> @jacobwoffenden Can you help provide me with repro steps (ie. dev container config)? Thanks!

it was an issue specifically on Ubuntu 22.04

```json
{
  "image": "mcr.microsoft.com/devcontainers/base:ubuntu",
  "features": {
    "ghcr.io/devcontainers/features/common-utils:2": { "configureZshAsDefaultShell": true },
    "ghcr.io/devcontainers/features/docker-in-docker:2": {}
  }
}
```

**Comment by mpriscella** [comment]
I also ran into this when using the `jammy` base image variant. It works on `bullseye`.

**Comment by jacobwoffenden** [comment]
proposing to switching from using pip to install, to Docker's instructions https://docs.docker.com/compose/install/standalone/ 

https://github.com/devcontainers/features/compare/main...jacobwoffenden:devcontainer-features:use-compose-v2

**Comment by jacobwoffenden** [comment]
interestingly docker/moby-compose is installed here too https://github.com/devcontainers/features/blob/main/src/docker-in-docker/install.sh#L235-L239

**Comment by samruddhikhandale** [comment]
Thanks for providing the dev container config, I am now able to reproduce the error. 

> 30.41 curl failed to verify the legitimacy of the server and therefore could not
> 30.41 establish a secure connection to it. To learn more about this situation and
> 30.41 how to fix it, please visit the web page mentioned above.

> I'm just preparing a PR, but was caught out by https://bugs.launchpad.net/ubuntu/+source/curl/+bug/2028170

Looks like the failures are caused by the upstream `curl` package which is available in the jammy feeds. 

> This bug was fixed in the package curl - 7.81.0-1ubuntu1.13
> It should appear in -security and get automatically copied to -updates next time the publisher runs, probably within the next half-hour or so.

From https://bugs.launchpad.net/ubuntu/+source/curl/+bug/2028170, looks like they have fixed the bug and should be available soon. 

> https://github.com/devcontainers/features/compare/main...jacobwoffenden:devcontainer-features:use-compose-v2

Thanks @jacobwoffenden, we appreciate all the help. Would you be willing to open the PR on this repo? As we are switching the  `dockerDashComposeVersion`'s default to `v2`, we'd need to bump the major version of the Feature. Hence, we need to be careful with the migration process as it's a breaking change.

**Comment by samruddhikhandale** [comment]
Update to `curl` package is now available, and I was able to successfully create the dev container with https://github.com/devcontainers/features/issues/619#issuecomment-1642503340

Opened https://github.com/devcontainers/features/issues/620 to track v2 for docker compose versioning.

Closing as completed, feel free to reopen if needed.
